From 044210c5296d294ce7ff9f888e343b9dfd612395 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Wed, 8 Apr 2020 23:52:24 -0400
Subject: [PATCH 1/6] Mark CGO tests with DependsOnCGO

  - issue10607 is only valid when running with CGO
  - TestPIESize requires CGO
  - TestVectoredHandlerDontCrashOnLibrary requires CGO
---
 src/cmd/link/elf_test.go           | 1 +
 src/runtime/signal_windows_test.go | 1 +
 test/fixedbugs/issue10607.go       | 2 +-
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/cmd/link/elf_test.go b/src/cmd/link/elf_test.go
index 88048ed2c5..087932d682 100644
--- a/src/cmd/link/elf_test.go
+++ b/src/cmd/link/elf_test.go
@@ -226,6 +226,7 @@ func main() {
 
 func TestPIESize(t *testing.T) {
 	testenv.MustHaveGoBuild(t)
+	testenv.MustHaveCGO(t)
 	if !sys.BuildModeSupported(runtime.Compiler, "pie", runtime.GOOS, runtime.GOARCH) {
 		t.Skip("-buildmode=pie not supported")
 	}
diff --git a/src/runtime/signal_windows_test.go b/src/runtime/signal_windows_test.go
index 9748403412..70e55638d7 100644
--- a/src/runtime/signal_windows_test.go
+++ b/src/runtime/signal_windows_test.go
@@ -21,6 +21,7 @@ func TestVectoredHandlerDontCrashOnLibrary(t *testing.T) {
 		t.Skip("this test can only run on windows/amd64")
 	}
 	testenv.MustHaveGoBuild(t)
+	testenv.MustHaveCGO(t)
 	testenv.MustHaveExecPath(t, "gcc")
 	testprog.Lock()
 	defer testprog.Unlock()
diff --git a/test/fixedbugs/issue10607.go b/test/fixedbugs/issue10607.go
index 6f4717d820..1f2edc8c3b 100644
--- a/test/fixedbugs/issue10607.go
+++ b/test/fixedbugs/issue10607.go
@@ -1,4 +1,4 @@
-// +build linux,!ppc64,!riscv64
+// +build linux,!ppc64,!riscv64,cgo
 // run
 
 // Copyright 2015 The Go Authors. All rights reserved.
-- 
2.24.2 (Apple Git-127)

